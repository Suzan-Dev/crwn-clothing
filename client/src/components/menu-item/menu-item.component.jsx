import React from 'react';
import { useNavigate } from 'react-location';
import './menu-item.style.scss';

export const MenuItem = ({ title, imageUrl, size, linkUrl }) => {
  const navigate = useNavigate();

  return (
    <div
      className={`menu-item ${size}`}
      onClick={() => navigate({ to: `${linkUrl}` })}
    >
      <div
        className="background-image"
        style={{ backgroundImage: `url(${imageUrl})` }}
      />
      <div className="content">
        <h1 className="title">{title}</h1>
        <span className="subtitle">SHOP NOW</span>
      </div>
    </div>
  );
};

export default MenuItem;
