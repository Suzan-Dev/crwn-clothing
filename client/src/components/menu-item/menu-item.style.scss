.menu-item {
  min-width: 30%;
  height: 240px;
  flex: 1 1 auto;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 7.5px 15px;
  overflow: hidden;

  &:hover {
    cursor: pointer;

    & .background-image {
      transform: scale(1.1);
      transition: transform 6s cubic-bezier(0.25, 0.45, 0.45, 0.95);
    }

    & .content {
      opacity: 0.9;
    }
  }

  & .background-image {
    height: 100%;
    width: 100%;
    background-position: center center;
    background-size: cover;
  }

  &.large {
    height: 240px;

    @media screen and (min-width: 800px) {
      height: 390px;
    }
  }

  &:first-child {
    margin-right: 7.5px;
  }

  &:last-child {
    margin-left: 7.5px;
  }

  .content {
    height: 90px;
    padding: 0 25px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background-color: white;
    opacity: 0.7;
    position: absolute;

    .title {
      font-weight: bold;
      margin-bottom: 6px;
      font-size: 22px;
      color: #4a4a4a;
      text-transform: uppercase;
    }

    .subtitle {
      font-size: 16px;
      font-weight: bold;
      color: #727272;
    }
  }
}
