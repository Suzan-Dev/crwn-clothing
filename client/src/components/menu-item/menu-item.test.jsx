import React from 'react';
import { shallow } from '../../setupTests';
import { MenuItem } from './menu-item.component';

describe('Testing: MenuItem component!', () => {
  const mockTitle = 'hats';
  const mockImageUrl = 'testImage';
  const mockSize = 'large';
  const mockLinkUrl = '/hats';
  const mockHistory = {
    push: jest.fn(),
  };
  const mockMatch = {
    path: '/shop',
  };

  const mockProps = {
    title: mockTitle,
    imageUrl: mockImageUrl,
    size: mockSize,
    linkUrl: mockLinkUrl,
    history: mockHistory,
    match: mockMatch,
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  test('should render MenuItem component.', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render title correctly.', () => {
    expect(wrapper.find('.title').text()).toBe(mockTitle);
  });

  test('should call history.push with the right string when MenuItem is clicked.', () => {
    wrapper.find('.menu-item').simulate('click');
    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.path}${mockLinkUrl}`);
  });
});
