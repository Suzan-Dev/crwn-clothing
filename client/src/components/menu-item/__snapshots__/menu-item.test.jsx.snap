// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: MenuItem component! should render MenuItem component. 1`] = `
"<div className=\\"menu-item large\\" onClick={[Function: onClick]}>
  <div className=\\"background-image\\" style={{...}} />
  <div className=\\"content\\">
    <h1 className=\\"title\\">
      hats
    </h1>
    <span className=\\"subtitle\\">
      SHOP NOW
    </span>
  </div>
</div>"
`;
